### ISSUE TITLE
question about q-former

### ISSUE THREAD

**Issue by peggyxpxu** [X-LANCE/SLAM-LLM#127]
Hi sir:
If i want to use q-former for projector in acc audiocaps, the length of the audio encoder Placeholder should set to 64?

**Comment by cwx-worst-one** [comment]
Yes, when setting the query length for q-former, it follows the configuration used in BLIP, which is set to 64. You can also customize the query length by modifying [this line](https://github.com/X-LANCE/SLAM-LLM/blob/683122402391806512a9d13febe8a952bb7c406e/src/slam_llm/models/projector.py#L61).

With this setup, you can refer to [this code](https://github.com/X-LANCE/SLAM-LLM/blob/683122402391806512a9d13febe8a952bb7c406e/src/slam_llm/datasets/audio_dataset.py#L119) to set the parameter `fix_length_audio=query_length (e.g., 64)` in the config, ensuring that the placeholder length matches the number of queries.

**Comment by peggyxpxu** [comment]
> Yes, when setting the query length for q-former, it follows the configuration used in BLIP, which is set to 64. You can also customize the query length by modifying [this line](https://github.com/X-LANCE/SLAM-LLM/blob/683122402391806512a9d13febe8a952bb7c406e/src/slam_llm/models/projector.py#L61).
> 
> With this setup, you can refer to [this code](https://github.com/X-LANCE/SLAM-LLM/blob/683122402391806512a9d13febe8a952bb7c406e/src/slam_llm/datasets/audio_dataset.py#L119) to set the parameter `fix_length_audio=query_length (e.g., 64)` in the config, ensuring that the placeholder length matches the number of queries.

Thanks, i use q-former with whisper encoder to train, but the loss of the model is much larger than i use linear with  whisper encoder. 
Have you ever done similar experiments?

**Comment by peggyxpxu** [comment]
I carefully analyzed the test results and found that the illusion of the model was very serious when using q-former.

**Comment by cwx-worst-one** [comment]
Yes, I have experimented with q-former, and I found that the train/validation loss is higher compared to using linear layers. This might be causing the model to exhibit hallucinations during training. However, I haven't done extensive hyperparameter tuning yet. You could try adjusting the query length and other parameters for further experimentation.
